Mathlib/RingTheory/MvPolynomial/Symmetric.lean

```diff
@@ -281,17 +294,59 @@ theorem degrees_esymm [Nontrivial R] (n : ℕ) (hpos : 0 < n) (hn : n ≤ Fintyp
 
 end ElementarySymmetric
 
+section CompleteHomogeneousSymmetric
+
+open Finset Multiset Sym
+
+variable [DecidableEq σ] [DecidableEq τ]
+
+/-- The `n`th complete homogeneous symmetric `MvPolynomial σ R`. -/
+def hsymm (n : ℕ) : MvPolynomial σ R := ∑ s : Sym σ n, (s.1.map X).prod
+
+lemma hsymm_def (n : ℕ) : hsymm σ R n = ∑ s : Sym σ n, (s.1.map X).prod := rfl
+
+def hsymm_mu {n : ℕ} (μ : Nat.Partition n) : MvPolynomial σ R :=
+  Multiset.prod (μ.parts.map (hsymm σ R))
+
+@[simp]
+theorem hsymm_zero : hsymm σ R 0 = 1 := by
+  simp only [hsymm, univ_unique, eq_nil_of_card_zero, val_eq_coe, Sym.coe_nil, Multiset.map_zero,
+    prod_zero, sum_const, Finset.card_singleton, one_smul]
+
+@[simp]
+theorem hsymm_one : hsymm σ R 1 = ∑ i, X i := by
+  simp only [hsymm, univ_unique]
+  symm
+  apply Fintype.sum_equiv oneEquiv
+  intro i
+  simp only [oneEquiv_apply, Multiset.map_singleton, Multiset.prod_singleton]
+
+theorem map_hsymm (n : ℕ) (f : R →+* S) : map f (hsymm σ R n) = hsymm σ S n := by
+  simp_rw [hsymm, map_sum, ← Multiset.prod_hom']
+  simp only [val_eq_coe, map_X]
+
+theorem rename_hsymm (n : ℕ) (e : σ ≃ τ) : rename e (hsymm σ R n) = hsymm τ R n := by
+  simp_rw [hsymm, map_sum, ← prod_hom', rename_X]
+  apply Fintype.sum_equiv (equivCongr e)
+  simp only [val_eq_coe, equivCongr_apply, Sym.coe_map, Multiset.map_map, Function.comp_apply,
+    implies_true]
+
+theorem hsymm_isSymmetric (n : ℕ) : IsSymmetric (hsymm σ R n) := rename_hsymm _ _ n
+
+end CompleteHomogeneousSymmetric
+
 section PowerSum
 
 open Finset
 
-variable (σ R) [CommSemiring R] [Fintype σ] [Fintype τ]
-
 /-- The degree-`n` power sum -/
 def psum (n : ℕ) : MvPolynomial σ R := ∑ i, X i ^ n
 
 lemma psum_def (n : ℕ) : psum σ R n = ∑ i, X i ^ n := rfl
 
+def psum_mu {n : ℕ} (μ : Nat.Partition n) : MvPolynomial σ R :=
```

Comment:
```suggestion
def psum_mu {n : ℕ} (μ : n.Partition) : MvPolynomial σ R :=
```
(same elsewhere)

You're making definitions `psum_mu`, `esymm_mu` and `hsymm_mu` without proving any theorems about them. Can you think of some theorems to prove? (Also, by naming convention these should be `psumMu` etc.)

Reply:
I planned to implement the monomial and Schur bases, and then prove some results about base changes and so on. I don't have a precise plan at the moment though.

Reply:
You can start by proving all the super trivial stuff: what happens when `n=0`, when `σ` is empty and stuff like that? It is perfectly fine (and even a good idea) to defer "serious" stuff to a later PR.